lib/loaders/api/loader_interface.js

```diff
@@ -0,0 +1,32 @@
+import { toKey } from "lib/helpers"
+
+const encodeStaticPath = (path, globalParams, params) => {
+  return toKey(path, Object.assign({}, globalParams, params))
+}
+
+const encodeDynamicPath = (pathGenerator, globalParams, id, params) => {
+  return encodeStaticPath(pathGenerator(id), globalParams, params)
+}
+
+/**
+ * This implements the common interface for producing data loaders for each api/path, regardless of authentication.
+ *
+ * If `path` is a string, then the loader will accept only query params as its parameters. For example:
+ *
+ *    const artworksLoader = gravityLoader("artworks")
+ *    artworksLoader({ page: 42 })
+ *
+ * If `path` is a function, then the loader accepts an ID _and_ query params as its parameters. For example:
+ *
+ *    const artistArtworksLoader = gravityLoader(id => `artist/${id}/artworks`)
```

Comment:
Was thinking `gravityLoader(artist/${id}/artworks)` but that's worse! Nevermind, this is nice.